pennylane/qchem/vibrational/taylor_ham.py

```diff
@@ -0,0 +1,704 @@
+                ): 0.00017035749999999995,
+            }
+        )
+    """
+    if is_local:
+        start_deg = 2
+    else:
+        start_deg = 3
+
+    harm_pot = _taylor_harmonic(coeffs, freqs)
+    ham = _taylor_anharmonic(coeffs, start_deg) + harm_pot
+    kin_ham = _taylor_kinetic(coeffs, freqs, is_local, uloc)
+    ham += kin_ham
+    return ham.normal_order()
+
+
+def taylor_hamiltonian(
+    pes, max_deg=4, min_deg=3, mapping="binary", n_states=2, ps=False, wire_map=None, tol=None
+):
+    """Return Taylor vibrational Hamiltonian.
+
+    Args:
+        pes (VibrationalPES): object containing the vibrational potential energy surface data
+        max_deg (int): maximum degree of Taylor form polynomial
+        min_deg (int): minimum degree of Taylor form polynomial
+        mapping (str): Mapping used to map to qubit basis. Input values can be ``"binary"`` or ``"unary"``.
+            Default is `"binary"`.
+        n_states(int): maximum number of allowed bosonic states
+        ps (bool): Flag to return the result as a PauliSentence instead of an
+            operator. Defaults to ``False``.
+        wire_map (dict): A dictionary defining how to map the states of
+            the Bose operator to qubit wires. If ``None``, integers used to
+            label the bosonic states will be used as wire labels. Defaults to ``None``.
+        tol (float): tolerance for discarding the imaginary part of the coefficients
+
+    Returns:
+        BoseSentence: the bosonic form of the Taylor Hamiltonian
+
+    **Example**
+
+    >>> pes_onemode = np.array([[0.309, 0.115, 0.038, 0.008, 0.000, 0.006, 0.020, 0.041, 0.070]])
+    >>> pes_twomode = np.zeros((1, 1, 9, 9))
+    >>> dipole_onemode = np.zeros((1, 9, 3))
+    >>> gauss_weights=np.array([3.96e-05, 4.94e-03, 8.85e-02,
+                        4.33e-01, 7.20e-01, 4.33e-01,
+                        8.85e-02, 4.94e-03, 3.96e-05])
+    >>> grid = np.array([-3.19, -2.27, -1.47, -0.72,  0.0,  0.72,  1.47,  2.27,  3.19])
+    >>> pes_object = qml.qchem.VibrationalPES(
+            freqs=np.array([0.025]),
+            grid=grid,
+            uloc=np.array([[1.0]]),
+            gauss_weights=gauss_weights,
+            pes_data=[pes_onemode, pes_twomode],
+            dipole_data=[dipole_onemode],
+            localized=True,
+            dipole_level=1,
+        )
+    >>> qml.qchem.taylor_hamiltonian(pes_object, 4, 2)
+    (
+        -0.003833496032473659 * X(0)
+        + (0.0256479442871582+0j) * I(0)
+        + (-0.013079509779221888+0j) * Z(0)
+    )
+    """
+    mapping.lower().strip()
+    coeffs_arr = taylor_coeffs(pes, max_deg, min_deg)
+    bose_op = taylor_bosonic(coeffs_arr, pes.freqs, is_local=pes.localized, uloc=pes.uloc)
+
+    if mapping not in ["binary", "unary"]:
+        raise ValueError(
+            f"Specified mapping {mapping}, is not found. Please use either 'binary' or 'unary' mapping."
+        )
```

Comment:
Let's move this at the very start, we don't want to do the whole Taylor calculation before figuring this out.